dockerfiles/theia/e2e/cypress/integration/theia/typescript.spec.ts

```diff
@@ -0,0 +1,108 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+context('TypeScript', () => {
+    before(() => {
+        cy.visit('http://localhost:3000');
+
+        // maybe it's possible to wait for an element being displayed/hidden
+        cy.wait(4000);
+    })
+
+    afterEach(() => {
+        cy.theiaCleanup();
+    });
+
+    // Create a typescript file and check we can have intellisense
+    it('Check Invalid Syntax', () => {
+
+        const FOLDER_NAME = 'typescripttest' + makeid();
+        const FILENAME = 'HelloWorld.ts';
+
+
+        // close any workspace
+        cy.theiaCommandPaletteClick('Close Workspace').then(() => {
+            const $el = Cypress.$('button.theia-button.main');
+            if ($el.length) {
+                cy.get('button.theia-button.main').should('exist').then(() => {
+                    cy.get('button.theia-button.main').click({ force: true });
+                });
+            }
+
+        }).then(() => {
+            // open /tmp
+            cy.get('#theia-top-panel').should('exist').then(() => {
+
+                cy.theiaCommandPaletteClick('Open...').then(() => {
+                    cy.get('.theia-LocationList').should('exist');
+                    cy.get('.theia-LocationList').select('file:///');
+                    cy.wait(2000);
+                    cy.get('.dialogContent .theia-TreeNodeSegment').should('exist');
+                    cy.get('.dialogContent .theia-TreeNodeSegment').contains('tmp').click({ force: true })
+                    cy.get('.dialogContent  .theia-TreeNodeSegment').contains('tmp').click({ force: true })
+                    cy.get('button.theia-button.main').click({ force: true });
+                })
+            });
+        }).then(() => {
+            // wait the refresh after workspace is opened
+            cy.wait(6000);
```

Comment:
Can we wait for some id to be present/visible?Because such behaviour is fragile.

Reply:
I agree with you but pb is that there is nothing on UI telling that command palette is ready or not.
When seeing the video recording of the test, UI is there but shortcut is not working for some time... if you have ideas